symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/ProviderLoader.java

```diff
@@ -0,0 +1,38 @@
+package com.symphony.bdk.core.util;
+
+import lombok.Generated;
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Slf4j
+@Generated
+@API(status = API.Status.INTERNAL)
+public class ProviderLoader {
```

Comment:
We could call this class `ServiceLookup` as it can load any kind of service. 

(I now remember my old time at school when we were developing [Netbeans plugins](http://bits.netbeans.org/dev/javadoc/org-openide-util-lookup/org/openide/util/Lookup.html))